### PR TITLE
WIP: Semantic conventions for Grafana's log/span attributes

### PR DESCRIPTION
**Pull request by marefr**
**What is this feature?**

Not sure about this yet, but something I've been thinking about for a while inspired by the [otel semantic conventions](https://opentelemetry.io/docs/specs/semconv/). Open for feedback, 👍 or 👎 . 

The main idea with o11ysomconv package is to provide all the common attributes that we're using all around the code base. For specific attributes to certain feature etc I imagine that each feature implements attributes internally to their feature code maybe, but just some random thoughts.

**Why do we need this feature?**

To allow developers to log and trace using consistent naming of attributes and help follow our [conventions and best practices](https://github.com/grafana/grafana/blob/main/contribute/backend/instrumentation.md). 

**Who is this feature for?**

Developers and operators.

**Which issue(s) does this PR fix?**:
Fixes #

**Special notes for your reviewer:**
- Don't put to much energy on the naming than can definitely be improved :) 
- Some examples included what it would look like

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by diegommm — COMMENTED** [review]
Love the idea! This would help so much to see the whole picture and have more homogeneous and predictable observability ("predictable" meaning we will need to think less about how something is called). And also makes it more scalable as more services can depend on the same conventions.

It would only mention inverting the import direction so that `o11ysemconv` is a completely independent package doing just one thing.

File: pkg/infra/o11ysemconv/definition.go [hunk 1]
```diff
@@ -0,0 +1,137 @@
+package o11ysemconv
```
[the remark this answers is not in the document]
Comment (marefr) [reply pkg/infra/o11ysemconv/definition.go:1]: Thanks. Maybe. Given

> For specific attributes to certain feature etc I imagine that each feature implements attributes internally to their feature code maybe, but just some random thoughts.

In these cases wouldn't make sense to add things to the tracing/log package.

[the remark this answers is not in the document]
Comment (marefr) [reply pkg/infra/o11ysemconv/definition.go:1]: So splitting the core definition support and the actual core attributes in separate packages might be better? Core definition would still depend on otel and the log attributes could potentially live in this package as well not adding a dependency on pkg/infra/log.

**Comment by diegommm** [comment]
Hi! Greetings from the Release Guild. Just letting you know I'm updating your PR's milestone to [11.2.x](https://github.com/grafana/grafana/milestone/529). If you need to see your changes in a previous version, please consider adding a backport label.

Thank you!

**Pull request closed without merging**